subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2162,6 +2166,10 @@ static int handle_request(struct zoap_packet *request,
 			zoap_header_set_code(out.out_zpkt,
 					     ZOAP_RESPONSE_CODE_NOT_ALLOWED);
 			r = 0;
+		} else if (r == -EEXIST) {
```

Comment:
NOTE: We should probably return BAD_REQUEST for other items and we should have a set of defined error codes to use for those cases.